modules/combined/test/tests/3d-mortar-projection-tolerancing/test.i

```diff
@@ -0,0 +1,228 @@
+    type = ADMatHeatSource
+    variable = 'T_K'
+    material_property = radial_source
+    block = 'pellet_inner pellet_outer '
+    use_displaced_mesh = true
+  []
+[]
+
+[Debug]
+  show_var_residual_norms = TRUE
+[]
+
+[BCs]
+  [mirror_z]
+    type = ADDirichletBC
+    variable = disp_z
+    boundary = 'mirror_innerp mirror_outerp  '
+    value = 0
+  []
+  [mirror_x]
+    type = ADDirichletBC
+    variable = disp_x
+    boundary = 'mirror_innerp mirror_outerp  '
+    value = 0
+  []
+  [mirror_y]
+    type = ADDirichletBC
+    variable = disp_y
+    boundary = 'mirror_innerp mirror_outerp  '
+    value = 0
+  []
+[]
+
+[Materials]
+  [pellet_properties]
+    type = ADGenericConstantMaterial
+    prop_names = 'density  thermal_conductivity specific_heat'
+    prop_values = '3.3112e3  34 1.2217e3'
+    block = 'pellet_inner pellet_outer'
+  []
+  [pulse_shape_linear]
+    type = ADGenericFunctionMaterial
+    prop_values = '5e10*max(11455*(t)/7,1e-9)'
+    prop_names = 'radial_source'
+    output_properties = 'radial_source'
+    block = 'pellet_inner pellet_outer'
+    use_displaced_mesh = false
+  []
+  [strain]
+    type = ADComputeSmallStrain
+    displacements = 'disp_x disp_y disp_z'
+    eigenstrain_names = eigenstrain #nameS!
+    block = 'pellet_inner pellet_outer '
+  []
+  [thermal_strain]
+    type = ADComputeThermalExpansionEigenstrain
+    stress_free_temperature = ${stress_free_temperature}
+    thermal_expansion_coeff = ${thermal_expansion_coeff}
+    eigenstrain_name = eigenstrain
+    block = 'pellet_inner pellet_outer '
+  []
+  [elasticity]
+    type = ADComputeIsotropicElasticityTensor
+    youngs_modulus = 3.306e11
+    poissons_ratio = 0.329
+  []
+  [stress]
+    type = ADComputeLinearElasticStress
+    block = 'pellet_inner pellet_outer '
+  []
+[]
+
+[Contact]
+  [pellet]
+    primary = void_pellet_0
+    secondary = void_pellet_1
+    model = frictionless
+    penalty = 1e6
+    formulation = mortar
+    c_normal = 1e6
+    correct_edge_dropping = true
+  []
+[]
+
+[UserObjects]
+  [conduction]
+    type = GapFluxModelConduction
+    temperature = T_K
+    boundary = 'void_pellet_0 void_pellet_1    '
+    gap_conductivity = 0.4
+    use_displaced_mesh = true
+  []
+  [rad_pellet]
+    type = GapFluxModelRadiation
+    temperature = T_K
+    boundary = void_pellet_0
+    primary_emissivity = 0.37
+    secondary_emissivity = 0.37
+    use_displaced_mesh = true
+  []
+[]
+
+[Constraints]
+  [gap_pellet]
+    type = ModularGapConductanceConstraint
+    variable = lm_pellet
+    secondary_variable = T_K
+    primary_boundary = 'void_pellet_0'
+    primary_subdomain = pellet_primary_subdomain
+    secondary_boundary = 'void_pellet_1'
+    secondary_subdomain = pellet_secondary_subdomain
+    gap_flux_models = 'conduction rad_pellet' #closed_pellet
+    gap_geometry_type = 'CYLINDER'
+    cylinder_axis_point_1 = '0 0 0'
+    cylinder_axis_point_2 = '0 0 1'
+    use_displaced_mesh = true
+    quadrature = SECOND
```

Comment:
What's this parameter doing? I don't recall it.

Reply:
It may be a remnant of the `quadrature = true` option from the older boundary condition.

Reply:
I set `quadrature=SECOND` because the description provided with this parameter
```
Default:DEFAULT
C++ Type:MooseEnum
Options:DEFAULT, FIRST, SECOND, THIRD, FOURTH
Controllable:No
Description:Quadrature rule to use on mortar segments. For 2D mortar DEFAULT is recommended. For 3D mortar, QUAD meshes are integrated using triangle mortar segments. While DEFAULT quadrature order is typically sufficiently accurate, exact integration of QUAD mortar faces requires SECOND order quadrature for FIRST variables and FOURTH order quadrature for SECOND order variables.
```

Reply:
It seems to me the `DEFAULT` should pick an integration rule according to the mesh. For a problem with first order elements, it would be 2 * LINEAR + 1 = 3. So `DEFAULT` may integrate with more points. Thanks for pointing that out.

Reply:
I think we should probably automate the selection of the integration rule and discourage (not forbid) controlling that parameter from the input file.